src/test_workflow/integ_test/distribution_tar.py

```diff
@@ -39,5 +39,9 @@ def start_cmd(self) -> str:
         return start_cmd_map[self.filename]
 
     def uninstall(self) -> None:
-        logging.info(f"Cleanup {self.work_dir}/* content after the test")
-        subprocess.check_call(f"rm -rf {self.work_dir}/*", shell=True)
+        logging.info(f"Cleanup {self.work_dir}/* content after the test and keep {self.install_dir}/logs for recording.")
+        subprocess.check_call(f"mv {self.install_dir}/logs {self.work_dir}", shell=True)
+        subprocess.check_call(f"rm -rf {self.install_dir}", shell=True)
```

Comment:
This is super dangerous. There's no guarantee that `self.install_dir` does not have spaces, or is blank. you'll be deleting all kinds of things. Replace all these system commands with Python code.

Reply:
Thanks for comments. Yeah I think current logics for uninstalling the cluster for tarball and windows is to force remove the entire `opensearch-{version}` folder. What do you think about we leave them alone and we only delete them along with the tmp workspace?

Reply:
I am good with any solution that doesn't call `rm -rf` in shell :)